File: gh-2516/src/main/java/com/example/gh2516/DepositTransaction.java
package com.example.gh2516;

import org.springframework.data.neo4j.core.schema.Node;
import org.springframework.data.neo4j.core.schema.Relationship;

@Node("DepositTransaction")
public class DepositTransaction extends WalletTransaction {

	@Relationship("FROM")
	public Wallet from;

	@Relationship("TO")
	public Wallet to;
}

File: gh-2516-kt/src/main/kotlin/com/example/gh2516kt/DepositTransaction.kt
package com.example.gh2516kt

import org.springframework.data.neo4j.core.schema.Node
import org.springframework.data.neo4j.core.schema.Relationship
import java.math.BigDecimal
import java.time.Instant

@Node("Deposit")
class DepositTransaction(
    id: String? = null,
    value: BigDecimal? = null,
    walletTransactionStatus: List<TransactionStatusUpdate>? = null,
    balance: WalletBalance? = null,
    createdAt: Long? = Instant.now().toEpochMilli(),
    updatedAt: Long? = null,

    @Relationship("FROM")
    val from: Wallet? = null,

    @Relationship("TO")
    val to: Wallet? = null,
) : WalletTransaction(id, value, walletTransactionStatus, balance, createdAt, updatedAt)
